File: PHPDataAdapters/StiFileAdapter.php
<?php
# Stimulsoft.Reports.JS
# Version: 2025.2.1
# Build date: 2025.03.20
# License: https://www.stimulsoft.com/en/licensing/reports
?>
<?php

namespace Stimulsoft\Adapters;

use Exception;
use Stimulsoft\Enums\StiDataType;
use Stimulsoft\StiDataResult;
use Stimulsoft\StiPath;

class StiFileAdapter extends StiDataAdapter
{

### Properties

    /** @var string The data type loaded by the data adapter. */
    protected $dataType = StiDataType::Text;

    /** @var StiPath Link to the created database connection driver. */
    protected $connectionLink;


### Methods

    protected function connect(): StiDataResult
    {
        $path = new StiPath($this->connectionString, $this->handler->checkFileNames);
        if ($path->filePath == null)
            return StiDataResult::getError("Data file '$this->connectionString' not found.")->getDataAdapterResult($this);

        $this->connectionLink = $path;

        return StiDataResult::getSuccess()->getDataAdapterResult($this);
    }

    public function getDataResult($filePath, $maxDataRows = -1): StiDataResult
    {
        $this->connectionString = $filePath;
        $this->process();

        $result = $this->connect();
        if ($result->success) {
            try {
                $result->data = file_get_contents($this->connectionLink->filePath);
                $result->dataType = $this->dataType;
            }
            catch (Exception $e) {
                $message = $e->getMessage();
                $result = StiDataResult::getError($message)->getDataAdapterResult($this);
            }
        }

        return $result;
    }
}

File: PythonDataAdapters/stimulsoft_data_adapters/StiFileAdapter.py
"""
Stimulsoft.Reports.JS
Version: 2025.2.4
Build date: 2025.05.19
License: https://www.stimulsoft.com/en/licensing/reports
"""

from urllib import response
import requests

from .classes.StiDataResult import StiDataResult
from .classes.StiPath import StiPath
from .enums.StiDataType import StiDataType
from .StiDataAdapter import StiDataAdapter


class StiFileAdapter(StiDataAdapter):
    
### Properties

    dataType = StiDataType.TEXT
    """The data type loaded by the data adapter."""

    connectionLink: StiPath = None
    """Link to the created database connection driver."""

### Methods

    def connect(self) -> StiDataResult:
        path = StiPath(self.connectionString)
        if path.filePath == None:
            return StiDataResult.getError(f"Data file '{self.connectionString}' not found.").getDataAdapterResult(self)

        self.connectionLink = path

        return StiDataResult.getSuccess().getDataAdapterResult(self)
    
    def getDataResult(self, filePath: str, maxDataRows: int = -1) -> StiDataResult:
        self.connectionString = filePath
        self.process()

        result = self.connect()
        if result.success:
            try:
                if self.connectionLink.fileUrl != None:
                    response = requests.get(self.connectionLink.fileUrl)
                    if response.status_code == 200:
                        result.data = response.content.decode()
                else:
                    with open(self.connectionLink.filePath, mode='r', encoding='utf-8') as file:
                        result.data = file.read()

                result.dataType = self.dataType.value
                
            except Exception as e:
                message = str(e)
                result = StiDataResult.getError(message).getDataAdapterResult(self)

        return result
    